src/common/ios/xcodeproj.ts

```diff
@@ -20,6 +20,7 @@ export class Xcodeproj {
     }
 
     public findXcodeprojFile(projectRoot: string): Q.Promise<string> {

```

Comment:
I think you should pass the absolute iosProjectPath to this method instead. I think that most ios methods only need to know the location of the ios project, so you'll be able to remove the projectRoot in most of them if you pass the iosProjectPath instead.